### PR TITLE
sg checks runner: fix data race

### PR DESCRIPTION
**Pull request by mrnugget**
Follow-up to #38878 - I didn't see that the categories themselves were
also executed concurrently.

Example failure here: https://buildkite.com/sourcegraph/sourcegraph/builds/161728#01821641-9c22-4e36-a8aa-419e7656521b

## Test plan

- Existing tests

### CODE REVIEW

**Review by burmudar — APPROVED** [review]
LGTM

File: dev/sg/internal/check/runner.go [hunk 1]
```diff
@@ -215,10 +215,26 @@ func (r *Runner[Args]) runAllCategoryChecks(ctx context.Context, args Args) *run
...
+		updateCategorySkipped = func(i int, err error) {
+			progressMu.Lock()
+			defer progressMu.Unlock()
+
+			progress.StatusBarCompletef(i, "Category skipped: %s", err.Error())
+		}
+		updateCategoryCompleted = func(i int) {
+			progressMu.Lock()
+			defer progressMu.Unlock()
+			progress.StatusBarCompletef(i, "Done!")
```
Comment (bobheadxi) [inline dev/sg/internal/check/runner.go:232]: I wonder if we can make `progress` itself thread-safe 🥲

Comment (mrnugget) [reply dev/sg/internal/check/runner.go:232]: Yeah, probably.

Comment (mrnugget) [reply dev/sg/internal/check/runner.go:232]: Turns out it is: https://github.com/sourcegraph/sourcegraph/blob/9aaf2906f906da5a02feaf4fd1f536b586554335/lib/output/progress_with_status_bars_tty.go#L108-L161

But that doesn't help the race detector, I guess?

**Pull request merged**